File: golang/lc00001-lc00100/lc00002/add_two_numbers.go
package lc00002

// ListNode 结点
type ListNode struct {
	Val  int
	Next *ListNode
}

func addTwoNumbers(l1 *ListNode, l2 *ListNode) *ListNode {
	first := &ListNode{l1.Val + l2.Val, nil}
	temp := first
	for l1.Next != nil || l2.Next != nil || temp.Val > 9 {
		temp.Next = &ListNode{temp.Val / 10, nil}
		temp.Val = temp.Val % 10
		temp = temp.Next
		if l1.Next != nil {
			temp.Val += l1.Next.Val
			l1 = l1.Next
		}
		if l2.Next != nil {
			temp.Val += l2.Next.Val
			l2 = l2.Next
		}
	}
	return first
}

File: php/lc00001-lc00100/lc00002/AddTwoNumbers.php
<?php

namespace leetcode\lc00002;

class AddTwoNumbers
{
    /**
     * @param ListNode $l1
     * @param ListNode $l2
     * @return ListNode
     */
    public function addTwoNumbers($l1, $l2)
    {
        $first = new ListNode($l1->val + $l2->val, null);
        $temp = $first;
        while ($l1->next || $l2->next || $temp->val > 9) {
            $temp->next = new ListNode(intval($temp->val / 10), null);
            $temp->val = $temp->val % 10;
            $temp = $temp->next;
            if ($l1->next !== null) {
                $temp->val += $l1->next->val;
                $l1 = $l1->next;
            }
            if ($l2->next !== null) {
                $temp->val += $l2->next->val;
                $l2 = $l2->next;
            }
        }
        return $first;
    }
}
